modules/drools/src/test/java/org.mule.module.drools/DroolsAssertBehaviorTestCase.java

```diff
@@ -37,16 +39,27 @@ public void setUp() throws Exception
     {
         when(rules.getResource()).thenReturn("rulesFile.drl");
         drools.setMessageService(messageService);
+    }
+
+    @Test
+    public void testEqualityAssertBehaviour() throws Exception
+    {
         DroolsSessionData sessionData = (DroolsSessionData) drools.createSession(rules);
         when(rules.getSessionData()).thenReturn(sessionData);
+        Object handle1 = drools.assertFact(rules, new TestFact("idTest", "descriptionTest"));
+        Object handle2 = drools.assertFact(rules, new TestFact("idTest", "descriptionTest"));
+        assertThat(handle1, is(handle2));
     }
 
     @Test
-    public void testMemoryLeakCausedBySaveDuplicatedObjects() throws Exception
+    public void testIdentityAssertBehaviour() throws Exception
     {
+        clearProperty(USE_EQUALITY_ASSERT_BEHAVIOR);
```

Comment:
You cannot ensure that this method will be executed before the other one. The right way to test this is to use a parameterized test that receives the property value and a function to assert the configuration, and then use org.mule.tck.MuleTestUtils.testWithSystemProperty